### PR TITLE
Create a single promise in streamPromise.

### PR DESCRIPTION
**Pull request by CodeMan99**
I'm creating the pull request to expose a bug when the destination does not close (stdout & stderr).

### CODE REVIEW

**Comment by CodeMan99** [comment]
Note that this passes all tests at a91130c, but it clearly does not handle errors from source or transform streams at all.

File: index.js [hunk 1]
```diff
@@ -9,39 +9,41 @@ class StreamError extends Error {
...
-      const stream = streams[streams.length - 1];
-      if (stream === process.stdout || stream === process.stderr) {
-        accept();
+  const dest = streams[streams.length - 1];
+
+  return new Promise((resolve, reject) => {
+    // insert a PassThrough stream before dest to listen for finish instead?
+    if (dest.autoClose === false || dest.fd === 1 || dest.fd === 2) {
+      resolve(streams);
+      return;
```
Comment (CodeMan99) [inline index.js]: Taking out this `return` will not make the new test case pass. Promise _creation_ is synchronous. Even with using `Promise.race` as master is currently, the `resolve` (or accept) will be hit first. Once a promise constructor callback is used, any call to either callback is ignored.

**Comment by MartinKolarik** [comment]
I think this issue was resolved by #10 since we ignore events on the stdout/stderr streams and wait for other streams in the chain to finish (this is also how it worked before #8 and I changed it back now).

If you agree it's ok now, update the PR to only include the additional test and I'll merge it.

**Comment by CodeMan99** [comment]
@MartinKolarik I pushed a fix that still creates a single promise per pipeline (`streams` array) instead of one promise per stream.

If you like the change, great. If not, just cherry-pick b7125a1 instead. That commit is only the new test.

**Comment by MartinKolarik** [comment]
Ok, I think I'll cherry-pick just the test for now and check this together with #2 later.

**Pull request closed without merging**